### ISSUE TITLE
Additional price series return format

### ISSUE THREAD

**Issue by dcwtx** [hydrosquall/tiingo-python#85]
Would it be useful to add an option to receive a pandas series as output?

If pandas were an output option, it would become possible to add a list of symbols, and return them all in a data frame.  At present it appears that you can only request prices for one symbol at a time.  Would there be a conflict with the json format if you returned multiple symbols (question belies my ignorance of json)?

**Comment by hydrosquall** [comment]
Hey @dcwtx - thanks for the suggestion!

> Would it be useful to add an option to receive a pandas series as output?

It would be possible, however it would mean adding pandas as a dependency to this library. This could be worthwhile if the productivity benefits for users were significant, I'm just a bit hesitant because installing Pandas/numpy can be troublesome particularly for Windows users who aren't using Anaconda due to all the C extensions involved.

Could you provide a small example of how you might expect the returned data to be shaped, so I can better understand the request? The JSON format is very flexible, so I think it could accommodate whichever shape you are thinking :).

Even if we don't end up adding this to the library core library, perhaps we can add an example to the documentation of what you're describing, namely how to get the prices for multiple symbols from 1 function call, and load them into a dataframe. 

It sounds like we'd put the existing `get_ticker_price` method in a loop, save all the results in a list, and do a bit of reshaping to put the results into a dataframe.

**Comment by dcwtx** [comment]
I don't use windows, so I'm not sure how the pandas wheels are delivered on that platform, but it surprises me that they have to be compiled - I thought they could and did upload pre-compiled wheels to the pip repository.

Anyway, let's say that you have a list of tickers: 
`tickers = ['GOOG', 'GE', 'FB', 'AMZN']`

If TiingoClient returns a data frame, then I'd use it something like this:
```
tc = TingoClient(config)
prices = tc.get_ticker_price(tickers, fmt='pandas')
returns = prices.pct_change()                                 
correls = returns.ewm(span=60).corr()                  
vol = returns.ewm(span=60).std()                          
weights = np.ones(4)*0.25                                      
portret = (returns*weights).cumsum()
portret.plot()                                                             
```
I could probably wrangle some other object into a data frame before doing all this but it would depend on the format of the object.  Here's how I've managed to do it with the current api:
```
tickers = [...]
pricedict = {}
for ticker in tickers:
    pricedict[ticker] = tc.get_ticker_price(ticker, startDate='...', endDate='...')

prices = pd.DataFrame()
for stock, price in pricedict.items():
    df = pd.DataFrame.from_dict(price)
    df.index = df['date']
    df.rename(index=str, columns={'adjClose': stock}, inplace=True)
    prices = pd.concat([prices, df[stock]], axis=1)
prices.index = pd.to_datetime(prices.index)
```
Then I'd proceed as above.

**Comment by dcwtx** [comment]
So following up here again... 

Pandas has a compiled wheel for win32 targeted to python 2.7, 3.5, and 3.6.  Maybe the compilation issue arises for a different python version?  https://pypi.org/project/pandas/#files

If instead you wanted to make pandas an optional install, the setup file could be modified per the discussion here - https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies.  And then if a user like me wanted to use pandas, he could `pip install tiingo[pandas]`.

**Comment by hydrosquall** [comment]
@dcwtx 

Hm, it was a while ago (4+ years) when I was trying to install Pandas on a windows computer, and I thought it might still be be tricky since the official Pandas docs recommend using [Anaconda](https://pandas.pydata.org/pandas-docs/stable/install.html), but perhaps things are better now (I use Mac/Linux as well now).

I like the idea of making Pandas an optional dependency, and would lean towards that in the interest of keeping this library relatively lightweight. However, from your code example (thanks for providing!), I can also see how adding a method like this would make it a lot easier for users to get started with this library.

Based on my reading of your code snippet, DataFrames seem like a convenient return format- would it be accurate to say that this is more usable than returning a list of Series?

**Comment by dcwtx** [comment]
Yes, as a user you’d much rather get a Dataframe instead of a list of Series. But that raises a good point, that in case the user asked for prices for a single symbol, you’d probably pass back a Series instead of a single-columned Dataframe, although the latter is perfectly valid. 

For me, and for others like me that program mainly for financial data analysis, pandas is the killer app for python. 

I will start a pull request later this week so we can look at how it would work.

**Comment by hydrosquall** [comment]
This has been fixed by #112 !
